src/lcm/input_processing/discrete_state_conversion.py

```diff
@@ -0,0 +1,244 @@
+from collections.abc import Callable
+from dataclasses import dataclass, field, make_dataclass
+
+import jax.numpy as jnp
+from dags.signature import with_signature
+from jax import Array
+
+from lcm.functools import all_as_kwargs
+from lcm.grids import DiscreteGrid
+from lcm.input_processing.util import (
+    get_gridspecs,
+)
+from lcm.typing import ParamsDict
+from lcm.user_model import Model
+
+
+@dataclass
+class DiscreteStateConverter:
+    """Converts between representations of discrete states and their parameters.
+
+    While LCM supports general discrete grids, internally, these are converted to
+    indices. This class provides functionality for converting between the internal
+    representation and the external representation.
+
+    Attributes:
+        converted_states: The names of the states that have been converted.
```

Comment:
```suggestion
        converted_grid_names: The names of the states that have been converted.
```
In the subsequent two descriptions, make clear that the dict keys are these names. Do we need `converted_grid_names` at all? Seems like there is just scope for confusion.